src/transaction/validation.rs

```diff
@@ -134,19 +150,35 @@ impl Input {
 }
 
 impl Output {
+    /// Validate the output of the transaction.
+    ///
+    /// This function is stateful - meaning it might validate a transaction during VM
+    /// initialization, but this transaction will no longer be valid in post-execution because the
+    /// VM might mutate the message outputs, producing invalid transactions.
     pub fn validate(&self, index: usize, inputs: &[Input]) -> Result<(), ValidationError> {
         match self {
             Self::Contract { input_index, .. } => match inputs.get(*input_index as usize) {
                 Some(Input::Contract { .. }) => Ok(()),
                 _ => Err(ValidationError::OutputContractInputIndex { index }),
             },
 
+            Self::Message { recipient, .. } if recipient != &Address::zeroed() => {
+                Err(ValidationError::OutputMessageRecipientNotZero { index })
+            }
+
+            Self::Message { amount, .. } if amount != &0 => {
+                Err(ValidationError::OutputMessageAmountNotZero { index })
+            }
```

Comment:
Why are zero-valued address and amount forbidden for message outputs? They're like variable outputs, they can be zero and if they're zero then they're not included in the UTXO set.

Reply:
We can't execute `SMO` if the recipient is non-zero. Isn't this an invalid tx? One that cannot ever have its output consumed by its respective opcode?

https://github.com/FuelLabs/fuel-specs/blob/e0d5d8aea3f6c344ca47776403accc58b67eeb01/specs/vm/instruction_set.md?plain=1#L1401

Reply:
I agree this validation check is necessary, but I'm confused why there isn't a similar check for OutputType.Variable to verify that amount is zero.